tests/test_test_tools/test_test_contract.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains tests for aea.test_tools.test_contract.BaseContractTestCase."""
+
+
+from pathlib import Path
+from unittest import mock
+
+import pytest
+from aea_ledger_fetchai import CosmosCrypto, FetchAIApi, FetchAICrypto, FetchAIFaucetApi
+
+from aea.test_tools.test_contract import BaseContractTestCase
+
+from tests.conftest import ROOT_DIR
+
+
+LEDGER_ID = "fetchai"
+CONTRACT_ADDRESS = "contract_address"
+
+
+class TestContractTestCase(BaseContractTestCase):
+    """Test case for BaseContractTestCase."""
+
+    path_to_contract = Path(ROOT_DIR, "tests", "data", "dummy_contract")
+
+    @classmethod
+    def setup(cls):
+        """Setup the test class."""
+        cls.ledger_identifier = LEDGER_ID
+
+        with mock.patch.object(
+            BaseContractTestCase,
+            "sign_send_confirm_receipt_multisig_transaction",
+            return_value="",
+        ):
+            with mock.patch.object(CosmosCrypto, "sign_transaction"):
+                with mock.patch.object(FetchAIApi, "get_deploy_transaction"):
+                    super().setup()
+
+    @classmethod
+    def finish_contract_deployment(cls):
+        """Finish contract deployment method."""
+        return CONTRACT_ADDRESS
+
+    def test_setup(self):
+        """Test the setup class method."""
+        assert self.ledger_identifier == LEDGER_ID
+        assert type(self.ledger_api) is FetchAIApi
+        assert self.ledger_api.identifier == LEDGER_ID
+
+        assert type(self.deployer_crypto) is FetchAICrypto
+        assert self.deployer_crypto.identifier == LEDGER_ID
+        assert type(self.item_owner_crypto) is FetchAICrypto
+        assert self.item_owner_crypto.identifier == LEDGER_ID
+
+        assert type(self.faucet_api) is FetchAIFaucetApi
+        assert self.item_owner_crypto.identifier == LEDGER_ID
+
+        assert self.contract_address == CONTRACT_ADDRESS
```

Comment:
Check the other class variables `fund_from_faucet`, `deployment_tx_receipt`

Reply:
Also cover 93 to 102 lines.

Reply:
Covered by lines 49, 56, 63